tests/test_push_pull_run_element_info.py

```diff
@@ -0,0 +1,161 @@
+            assert poa.return_value[0].udf['RE Useable Comment'] == 'AR: To much good yield'
+
+            assert poa.return_value[1].udf['RE Useable'] == 'yes'
+            assert poa.return_value[1].udf['RE Useable Comment'] == 'AR: Good yield'
+
+            assert poa.return_value[2].udf['RE Useable'] == 'no'
+            assert poa.return_value[2].udf['RE Useable Comment'] == 'AR: Failed and not needed'
+
+
+        patched_output_artifacts_per_sample = patch_output_artifact([
+                Mock(spec=Artifact, udf={'RE Yield Q30': 115, 'RE %Q30': 85, 'RE Review status': 'pass'}),
+                Mock(spec=Artifact, udf={'RE Yield Q30': 15, 'RE %Q30': 70, 'RE Review status': 'fail'}),
+            ]
+        )
+        with patched_output_artifacts_per_sample as poa:
+            self.epp.assess_sample(sample)
+            assert poa.return_value[0].udf['RE Useable'] == 'yes'
+            assert poa.return_value[0].udf['RE Useable Comment'] == 'AR: Good yield'
+
+            assert poa.return_value[1].udf['RE Useable'] == 'no'
+            assert poa.return_value[1].udf['RE Useable Comment'] == 'AR: Failed and not needed'
+
+
+
+class TestPushRunElementInfo(TestEPP):
+    def setUp(self):
+        self.patched_samples = patch.object(
+            PushRunElementInfo,
+            'samples',
+            new_callable=PropertyMock(return_value=[NamedMock(real_name='sample1')])
+        )
+        self.patched_process = patch.object(
+            PushRunElementInfo,
+            'process',
+            new_callable=PropertyMock(return_value=Mock(id='processid'))
+        )
+        self.patched_lims = patch.object(PushRunElementInfo, 'lims', new_callable=PropertyMock)
+        run_element = {
+            'run_element_id':'id',
+            'reviewed': 'pass',
+            'review_comments': 'alright',
+            'review_date': '12_02_2107_12:43:24'
+        }
+        self.patched_rest_get_doc = patch.object(PushRunElementInfo, 'get_documents', return_value=[run_element])
+        self.patched_rest_patch = patch('egcg_core.rest_communication.patch_entry')
+
+        self.patched_output_artifacts_per_sample = patch.object(
+            PushRunElementInfo,
+            'output_artifacts_per_sample',
+            return_value=[Mock(spec=Artifact, udf={'RE Id': 'id', 'RE Useable': 'no', 'RE Useable Comment': 'too bad'})]
+        )
+        self.epp = PushRunElementInfo(
+            'http://server:8080/a_step_uri',
+            'a_user',
+            'a_password',
+            self.log_file
+        )
+
+
+    def test_push(self):
+        with self.patched_lims as pl, self.patched_samples as ps, \
+                self.patched_rest_get_doc as prgd, self.patched_rest_patch as prp, \
+                self.patched_process as pp, \
+                self.patched_output_artifacts_per_sample as poa:
+            self.epp.run()
+            # Check that the run element was updated
+            prp.assert_any_call_with(
```

Comment:
This mock method does not exist, and raises an error in my dev environment. I don't understand how this passes in Travis.

Reply:
I guess it passed because you can call any method with any argument on a `Mock`.
Not sure why it raised an error in your env